### ISSUE TITLE
Error loading tests with regex in package.json

### ISSUE THREAD

**Issue by pbotsman** [kavod-io/vscode-jest-test-adapter#83]
**Describe the bug**
- [ ] test load missing suites/tests
- [x] test load errors
- [ ] test running errors

**To Reproduce**
(I am not sure if you will be able to reproduce the error, but hopefully this will help)
1. Have a package.json in the root of the project
2. Put jest configuration in a `package.json`:
```json
{
    ... // other project settings
    "jest": {
    "modulePaths": [
      "<rootDir>/build/js/"
    ],
    "setupTestFrameworkScriptFile": "<rootDir>/tests/js/testFrameworkScriptFile.js",
    "testRegex": ".*\\.js"  // In my case regex is a little bit more complicated, but I am facing an error even with this one 
  }
}
```
3. Put tests in the `tests/js/`
4. Try to reload tests
5. Get an error:
```
[2020-09-17 13:56:51.935] [ERROR] Error while reloading all tests. [TypeError: e.testRegex.some is not a function
	at /Users/pavel/.vscode/extensions/kavod-io.vscode-jest-test-adapter-0.7.2/dist/extension.js:425:2535079
	at Array.some (<anonymous>)
	at /Users/pavel/.vscode/extensions/kavod-io.vscode-jest-test-adapter-0.7.2/dist/extension.js:425:2534814
	at t.default.<anonymous> (/Users/pavel/.vscode/extensions/kavod-io.vscode-jest-test-adapter-0.7.2/dist/extension.js:425:2533881)
	at Generator.next (<anonymous>)
	at o (/Users/pavel/.vscode/extensions/kavod-io.vscode-jest-test-adapter-0.7.2/dist/extension.js:16:1418)]
```

**Expected behavior**
Tests are explored correctly and I can see them in the tests tab

**Desktop (please complete the following information):**
 - OS: OSX

**Additional context**
Here is a full output from the extension when I try to reload tests:
```
[2020-09-17 13:56:51.803] [INFO] Loading Jest tests...
[2020-09-17 13:56:51.805] [INFO] Loading Jest settings from /Users/pavel/projects/eb/package.json...
[2020-09-17 13:56:51.932] [INFO] More than one Jest test regex found. {
  jestVersionMajor: 23,
  configs: [
    {
      automock: false,
      browser: false,
      cache: true,
      cacheDirectory: '/var/folders/0_/6w94lk1171vc_fpwbjqx02nw0000gn/T/jest_dx',
      clearMocks: false,
      coveragePathIgnorePatterns: [Array],
      detectLeaks: false,
      detectOpenHandles: false,
      errorOnDeprecated: false,
      filter: null,
      forceCoverageMatch: [],
      globals: {},
      haste: [Object],
      moduleDirectories: [Array],
      moduleFileExtensions: [Array],
      moduleNameMapper: {},
      modulePathIgnorePatterns: [],
      modulePaths: [Array],
      name: '523afd3e38b2345a8f12a6a37c91da53',
      prettierPath: null,
      resetMocks: false,
      resetModules: false,
      resolver: null,
      restoreMocks: false,
      rootDir: '/Users/pavel/projects/eb',
      roots: [Array],
      runner: 'jest-runner',
      setupFiles: [Array],
      setupTestFrameworkScriptFile: '/Users/pavel/projects/eb/tests/js/testFrameworkScriptFile.js',
      skipFilter: false,
      snapshotSerializers: [],
      testEnvironment: 'jest-environment-jsdom',
      testEnvironmentOptions: {},
      testLocationInResults: false,
      testMatch: [],
      testPathIgnorePatterns: [Array],
      testRegex: '.*\\.js',
      testRunner: '/Users/pavel/projects/eb/node_modules/jest-cli/node_modules/jest-jasmine2/build/index.js',
      testURL: 'http://localhost',
      timers: 'real',
      transform: [Array],
      transformIgnorePatterns: [Array],
      watchPathIgnorePatterns: []
    }
  ]
}
[2020-09-17 13:56:51.933] [INFO] Force loading all tests...
[2020-09-17 13:56:51.933] [INFO] Loading Jest tests
[2020-09-17 13:56:51.935] [ERROR] Error while reloading all tests. [TypeError: e.testRegex.some is not a function
	at /Users/pavel/.vscode/extensions/kavod-io.vscode-jest-test-adapter-0.7.2/dist/extension.js:425:2535079
	at Array.some (<anonymous>)
	at /Users/pavel/.vscode/extensions/kavod-io.vscode-jest-test-adapter-0.7.2/dist/extension.js:425:2534814
	at t.default.<anonymous> (/Users/pavel/.vscode/extensions/kavod-io.vscode-jest-test-adapter-0.7.2/dist/extension.js:425:2533881)
	at Generator.next (<anonymous>)
	at o (/Users/pavel/.vscode/extensions/kavod-io.vscode-jest-test-adapter-0.7.2/dist/extension.js:16:1418)]
[2020-09-17 13:56:51.935] [INFO] Results:
[2020-09-17 13:56:51.935] [INFO] Finished loading Jest tests.
```

The extension says that `More than one Jest test regex found`, but I don't see any other regexes. 
I've tried to modify my regex from `package.json` and neither seem to work for me.
However, if I remove regex, the extension loads tests successfully (but more than I need, because I am using regex to exclude some portion of tests).

I am happy to provide any additional information if necessary.

Thanks!

**Comment by rossknudsen** [comment]
Hmm, I think the offending code is this:

![image](https://user-images.githubusercontent.com/2489915/93546556-2303bb80-f9b7-11ea-9fd3-da350eb5328f.png)

Notice that the type of `c.testRegex` is `string[]` however I suspect at runtime it is a `string`.  Which means the types are incorrect.  Can you try setting your `testRegex` to `['.*\\.js']` i.e. convert it to an actual array of strings?  I'd be interested to know if this is the case or not.

Note that the documentation for Jest [actually says that it should be a string or an array of strings](https://jestjs.io/docs/en/configuration#testregex-string--arraystring), so I'm guessing that the type definition is incorrect.  But I'd still be interested in finding out whether it works for you with the above config.  If you are feeling adventurous, you could raise a PR to fix this.

By the way, what version of Jest are you running (it may be that the types have changed).

**Comment by pbotsman** [comment]
If I change my `testRegex` to an array, then I am getting following error:
```
[2020-09-18 05:41:44.870] [ERROR] Error loading tests '● Validation Error:\n' +
  '\n' +
  '  Option "testRegex" must be of type:\n' +
  '    string\n' +
  '  but instead received:\n' +
  '    array\n' +
  '\n' +
  '  Example:\n' +
  '  {\n' +
  '    "testRegex": "(/__tests__/.*|(\\\\.|/)(test|spec))\\\\.jsx?$"\n' +
  '  }\n' +
  '\n' +
  '  Configuration Documentation:\n' +
  '  https://jestjs.io/docs/configuration.html\n' +
  '\n'
```

My jest version is `jest@25.2.3`

**Comment by rossknudsen** [comment]
This should be fixed in the latest version (0.7.4) but if you still experience issues, please reopen this.

**Comment by pbotsman** [comment]
Now it works, thank you!
